Scripts/Runtime/Data/Collections/UnsafeHeap.cs

```diff
@@ -0,0 +1,610 @@
+using Anvil.Unity.DOTS.Entities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    //NOTE - Taken from: https://github.com/Amarcolina/NativeHeap (MIT LICENCE)
+    //This is a lightly cleaned up version of NativeHeap from Amarcolina that removes
+    //any safety aspects so that it can be used inside a different Native Collection.
+    
+    /// <summary>
+    /// This is a basic implementation of the MinHeap/MaxHeap data structure.  It allows you
+    /// to insert objects into the container with a O(log(n)) cost per item, and it allows you
+    /// to extract the min/max from the container with a O(log(n)) cost per item.
+    /// 
+    /// This implementation provides the ability to remove items from the middle of the container
+    /// as well.  This is a critical operation when implementing algorithms like astar.  When an
```

Comment:
```suggestion
    /// as well.  This is a critical operation when implementing algorithms like A-Star.  When an
```

or A*
Up to you
https://en.wikipedia.org/wiki/A*_search_algorithm

Reply:
I didn't write this, Amarcolina did, but it was a quick change so done.